### PR TITLE
Remove deprecated `decimal_cols_as_float` in the ORC reader

### LINKED ISSUE #10129: [FEA] Remove the ORC reader option to read decimal columns as float
Now that all decimal types are available in C++ and Python, conversion to float should not be used as it leads to data loss. 

- [x] Deprecate the APIs - 22.04
- [ ] Remove the API and implementation - 22.06

### PR DESCRIPTION
**Pull request by vuule**
Closes #10129

### CODE REVIEW

File: python/cudf/cudf/io/orc.py [hunk 1]
```diff
@@ -285,18 +285,10 @@ def read_orc(
     skiprows=None,
     num_rows=None,
     use_index=True,
-    decimal_cols_as_float=None,
     timestamp_type=None,
     use_python_file_object=True,
     **kwargs,
 ):
-    """{docstring}"""
```
Comment (galipremsagar) [inline python/cudf/cudf/io/orc.py:293]: We'll need to keep the `docstring`

**Comment by vuule** [comment]
rerun tests

**Review by galipremsagar — APPROVED** [review]

**Review by bdice — APPROVED** [review]
Helping out with a review here since I reviewed the previous PR to deprecate this parameter. This looks great!

**Review by ttnghia — APPROVED** [review]

**Comment by vuule** [comment]
@gpucibot merge

**Pull request merged**